lib/libzfs/libzfs_util.c

```diff
@@ -596,27 +596,47 @@ zfs_strdup(libzfs_handle_t *hdl, const char *str)
  * Convert a number to an appropriately human-readable output.
  */
 void
-zfs_nicenum(uint64_t num, char *buf, size_t buflen)
+__zfs_nicenum(uint64_t num, char *buf, size_t buflen,
```

Comment:
I have the sense that adding a `zfs_nicetime()` and `zfs_nicenum10()` would be the more maintainable thing to do here as well.  You might collapse provide a common `zfs_nicenum_impl()` function which takes the `k_units` as an argument and wrap it with `zfs_nicenum` and `zfs_nicenum10`.